src/smk/api/route-planner.js

```diff
@@ -161,6 +160,16 @@ include.module( 'api.route-planner-js', [ 'jquery', 'util' ], function () {
 
                     // debugger
                 }
+                if (option.rangeKm > 0 && data.distance && option.rangeKm < data.distance) {
+                    var distanceKm = 0;
+                    for (var i = 0; i < data.route.length + 1; i += 1) {
+                        distanceKm += turf.distance(data.route[i], data.route[i+1], {units: "kilometers"});
+                        if (distanceKm >= option.rangeKm) {
+                            data.rangeLimit = data.route[i];
+                            break;
+                        }
+                    }
+                }
```

Comment:
I like. Seems like performance is fair (~2 s), even for long routes with longer ranges (750 km)